tests/rust-integration-tests/integration_test/src/tests/hostname/mod.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::{Context, Result};
+use oci_spec::runtime::{LinuxBuilder, ProcessBuilder, Spec, SpecBuilder};
+use test_framework::{Test, TestGroup, TestResult};
+
+use crate::utils::test_inside_container;
+
+fn create_spec(hostname: &str) -> Result<Spec> {
+    Ok(SpecBuilder::default()
+        .hostname(hostname)
+        .linux(
+            LinuxBuilder::default()
+                .readonly_paths(vec![
+                    "/proc/bus".to_string(),
+                    "/proc/fs".to_string(),
+                    "/proc/sys".to_string(),
+                ])
+                .build()
+                .expect("should build linux config"),
+        )
+        .process(
+            ProcessBuilder::default()
+                .args(vec!["runtimetest".to_string()])
+                .build()
+                .expect("can create process"),
+        )
+        .build()
+        .context("Failed to build hostname spec")?)
+}
+
+fn hostname_test() -> TestResult {
+    let cases = vec!["hostname-specific", ""];
+    for case in cases {
+        let spec = create_spec(case).expect("should create spec");
+        let test_result = test_inside_container(spec, &|_| {
+            // As long as the container is created, we expect the hostname to be determined
+            // by the spec, so nothing to prepare prior.
+            Ok(())
+        });
+        // fail fast
+        if let TestResult::Failed(e) = test_result {
+            return TestResult::Failed(e);
+        }
+    }
+    TestResult::Passed
+}
+
+pub fn get_hostname_test() -> TestGroup {
+    let mut test_group = TestGroup::new("set_host_name");
+    let hostname_test = Test::new("set_host_name_test", Box::new(hostname_test));
+    test_group.add(vec![Box::new(hostname_test)]);
+
+    test_group
+}
```

Comment:
It is preferable that we don't loop over test cases as in the original ones. Instead, please make two separate functions for each and add them to test group. It sometimes get pretty confusion in the original source (oci-tests) to see what tests exactly are there, and what are their expected outcomes. Also separating functions will make it easier if we ever want to modify the case later.

Also, after that we can simply return the result of `test_inside_container` instead of checking for error